### ISSUE TITLE
Fix the addition of a control without unrolling for circuits that consist of simple gates such as `X`.

### ISSUE THREAD

**Issue by Zoufalc** [Qiskit/qiskit-terra#4675]
### Information

- **Qiskit Terra version**:  0.15.0
- **Python version**:  3.7.3
- **Operating system**: macOS Catalina 10.15.1

### What is the current behavior?

Given a simple quantum circuit consisting, e.g., of a single X gate which is converted to a gate and then transformed into a controlled gate, the current method returns a composed gate that consists of multiple 2-qubit and 1-qubit gates instead of, e.g., a CX.

### Steps to reproduce the problem

```python
qc = QuantumCircuit(1)
qc.x(0)
controlled = QuantumCircuit(2)
controlled.append(qc.to_gate().control(), [0, 1])
print(controlled.decompose().draw(fold=-1))
```
produces
```
     ┌──────────┐                                                                                   ┌───────┐
q_0: ┤ U1(pi/2) ├──■─────────────────────■─────────────────────────────────■─────────────────────■──┤ U1(0) ├──■─────────────────■───────────────
     ├──────────┤┌─┴─┐┌───────────────┐┌─┴─┐┌───────────┐┌──────────────┐┌─┴─┐┌───────────────┐┌─┴─┐├───────┤┌─┴─┐┌───────────┐┌─┴─┐┌───────────┐
q_1: ┤ U1(pi/2) ├┤ X ├┤ U3(0,0,-pi/2) ├┤ X ├┤ U3(0,0,0) ├┤ U3(pi/2,0,0) ├┤ X ├┤ U3(-pi/2,0,0) ├┤ X ├┤ U1(0) ├┤ X ├┤ U3(0,0,0) ├┤ X ├┤ U3(0,0,0) ├
     └──────────┘└───┘└───────────────┘└───┘└───────────┘└──────────────┘└───┘└───────────────┘└───┘└───────┘└───┘└───────────┘└───┘└───────────┘
```

### What is the expected behavior?

Return a CX

### Suggested solutions

It seems as if the `operation.name` check in `add_control.py` cannot identify the respective gate as `X`, since the `operation.name` is set to `circuit__`.

**Comment by faisaldebouni** [comment]
I would like to give it a try if nobody else is working on this.

**Comment by kdk** [comment]
All yours, @faisaldebouni .

**Comment by Cryoris** [comment]
It seems this issue has been fixed as by-product in 8f717d90a235dfa1d23b2be5171d4c31224c4ab3.  However that was only a superficial fix by increasing the lookup table for gates we can efficiently control. The correct way would be to improve the synthesis of controlled operations, I'll open a new issue for that.
